### ISSUE TITLE
Per-route object based headers configuration

### ISSUE THREAD

**Issue by vejja** [Baroshem/nuxt-security#292]
It is not possible to define CSP with the `ContentSecurityPolicy` object type in the per-route configuration.

### Version

nuxt-security:  1.0.0-rc.4
nuxt:  3.8.0

### Reproduction Link

Stackblitz:
https://stackblitz.com/edit/nuxt-starter-pf1ssj?file=app.vue

### Steps to reproduce

Open Stackblitz on home page and observe console errors.

### What is actually happening?

Object format is not supported and CSP header has an invalid value:

<img width="940" alt="Capture d’écran 2023-11-15 à 16 00 39" src="https://github.com/Baroshem/nuxt-security/assets/7295259/9ebf83f6-08dc-46b9-9602-fda078b6edcd">

In addition, TypeScript complains about the `object` type:

<img width="771" alt="Capture d’écran 2023-11-15 à 16 09 02" src="https://github.com/Baroshem/nuxt-security/assets/7295259/72687eb2-1534-47ab-8f15-1c475f3400b7">

**Comment by Baroshem** [comment]
Hey @vejja 

I have modified the title and label to be an enhancement rather than a bug and I have also added it to the 1.1.0 milestone :)

**Comment by vejja** [comment]
Hey @Baroshem 

I have a PR for this which is almost ready. The logic is now that every security option is resolved on a per-route level with the radix router, and it works very well.

I was so happy with the result that I tried to extend the same logic to _all_ security options ! 
But this is more difficult than I thought initially :

1- Some options cannot be configured on a per-route level: this is the case of `removeLoggers` which removes all logs for the whole application, and I guess also `csrf` which installs the module globally.

2- Some options can be configured on per-route level, but implementation depends on design decision: this is the case of `rateLimiter`.

Say you have a limit of 10 requests/sec at the global level, but `/route1` has a limit of 5 req/s. What happens if you hit `index` 5 times, can you still hit `/route1` ? In other words, are the limits _cumulative_ or _independent_ ?
- Right now the design is that they are _cumulative_, because we record the number of hits _per ip_ and not _per route_. This sounds sensible but in this case we cannot have a `driver` option at the per-route level: there must be only one storage driver for the whole application
- Conversely, users might expect they are _independent_. The classic use case is a very high limit at the global level, but a much lower one on some specific `/api/protected` route. Currently, if someone hits normal routes at a high rate, `/api/protected` would always be denied, **even at the first hit** because other routes have been accessed before and the limit is cumulative right now.

3- Some options could be configured at the per-route level, but their current syntax overrides the functionality: this is the case of `basicAuth`, which already provides `include` and `exclude` options and these options can conflict with the logic of the radix resolver.

4- Some options might be configured at the per-route level, but implementation depends on the scope of protection. By _scope_, I mean: is the option only protecting HTML resources delivered to the browser, or also `/api` server routes ?

For instance I think you said in another thread that `basicAuth` was only supposed to work for HTML resources, and indeed the module naturally sounds like it is targeting the browser's security, and not the server's security. Conversely, a user can imagine that all `middlewares` are designed to ensure security of the server (which they are, in fact!).

**In conclusion**, I will submit a PR that only implements the functionality for CSP for now.

If you have some comments on this post, let me know. It might help me understand something that I got fundamentally wrong, or it might lead to extending additional options at the per-route level.

**Comment by Baroshem** [comment]
Hey! 

Thanks for extensive research!

Regarding the design decision. I will come back to you with it after I will be back from vacation as I want to give you a proper answer to that.

Regarding the object headers configuration for route rules. In the last paragraph you mentioned CSP. Does your solution work also for Strict Transport Policy and Permissions Policy?

Ideally this is what we would like to have there :)

**Comment by Baroshem** [comment]
Awesome thanks.

I just want to say that I really appreciate all the work you do! 💚

**Comment by vejja** [comment]
Thanks ! I'm very happy to contribute up to my capabilities - I'm also a user 😉
